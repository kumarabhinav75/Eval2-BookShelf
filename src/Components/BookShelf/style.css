.BookShelf {
  background-color: rgb(55, 55, 55);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.Logo {
  height: 30px;
  width: 100%;
  border-bottom: 5px solid white;
  text-align: center;
  padding: 10px;
}

.LogoName {
  font-size: 1.5rem;
  padding: 0 10px;
  background-color: rgb(55, 55, 55);
  color: white;
  display:flex;
  flex-direction: column;
  width: 300px;
  margin: auto;
}

.LogoText {
  background-color: rgb(55, 55, 55);
  padding: 0 25px;
  display: inline;
  font-size: 1.5rem;
  font-weight: bold;
  font-style: italic;
}
.LogoIcon {
  background-color: rgb(55, 55, 55);
  padding: 5px;
  display: inline;
  border: 1px solid white;
  border-radius: 30px;
  width: 30px;
  margin: auto;
  height: 30px;
  font-weight: bold;

}

.author-container {
  border: 5px solid darkorange;
  height: auto;
  width: 90%;
  border-radius: 30px;
  margin: 70px auto;
  text-align: center;
  display: flex;
  justify-content: space-evenly;
}

.author-container legend {
  background: white;
  color: darkorange;
  font-size: 1.4rem;
  padding: 10px;
  border-radius: 3px;
  font-style: italic;
  font-weight: bold;
}