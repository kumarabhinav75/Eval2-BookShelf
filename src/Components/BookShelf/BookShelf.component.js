import React, {Component} from 'react';
import AuthorShelf from '../AuthorShelf/AuthorShelf.component'

// import cardJSON from '../../mockData/index.json';

import './style.css';

const axios = require('axios');

class Main extends Component {
  state = {
    BooksJSON: [],
  }

  componentDidMount() {
    return axios.get('http://localhost:8080/books')
      .then((jsonObj) => {
        this.setState({BooksJSON: jsonObj.data});
        return jsonObj;
      });
  }

  render(){
    console.log(this.state.BooksJSON["J K Rowling"]);
    return (
      <div className="BookShelf">
        <div className="Logo">
          <span className="LogoName" >
            <div className="LogoIcon">BS</div>
            <div className="LogoText">The Book Shelf</div>
          </span>
        </div>
        <div className="author-shelfs">
          <fieldset className="author-container">
          <legend>J.K.Rowling</legend>
            <AuthorShelf books={this.state.BooksJSON["J K Rowling"]}/>
          </fieldset>
          <fieldset className="author-container">
            <legend>Sidney Sheldon</legend>
            <AuthorShelf books={this.state.BooksJSON["Sidney Sheldon"]}/>
          </fieldset>
        </div>

      </div>
    );
  }
}


export default Main;