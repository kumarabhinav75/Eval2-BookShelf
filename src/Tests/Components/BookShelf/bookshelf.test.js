import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import BookShelf from '../../../Components/BookShelf/BookShelf.component';

describe('BookShelf', () => {
  it ('Should match the snapshot', () => {
    const tree = renderer.create(<BookShelf />);
    expect(tree).toMatchSnapshot();
  });

  it ('should call componentDidMount once', () => {
    // console.log('TEST' , wrapper.instance().componentDidMount);
    const fakeComponentDidMount = jest.spyOn(BookShelf.prototype, 'componentDidMount');
    const wrapper = mount(<BookShelf />);
    expect(fakeComponentDidMount).toHaveBeenCalledTimes(1);
  });

})