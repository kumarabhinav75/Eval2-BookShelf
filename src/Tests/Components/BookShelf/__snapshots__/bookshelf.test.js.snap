// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf Should match the snapshot 1`] = `
<div
  className="BookShelf"
>
  <div
    className="Logo"
  >
    <span
      className="LogoName"
    >
      <div
        className="LogoIcon"
      >
        BS
      </div>
      <div
        className="LogoText"
      >
        The Book Shelf
      </div>
    </span>
  </div>
  <fieldset
    className="author-container"
  >
    <legend>
      J.K.Rowling
    </legend>
     
  </fieldset>
  <fieldset
    className="author-container"
  >
    <legend>
      Sidney Sheldon
    </legend>
     
  </fieldset>
</div>
`;
